scw-boot-tools/scw-update-server-state.c

```diff
@@ -1,116 +1,702 @@
+    t_request   request;
+    t_client    client;
+    size_t      length;
+    char        *path;
+    bool        ret;
+	e_method	method;
+	const char	*tab_str[2] = {0, 0};
+	int			i;
+	char		*parse;
+	int			status_code;
+
+	i = 0;
+	method = e_met_get;
+	parse = (char *)argv;
+	while ((tab_str[i] = strsep(&parse, "=")) != NULL && i < 2) {
+		i = i + 1;
+	}
+	if (tab_str[0] != NULL && tab_str[1] != NULL) {
+		if (tab_str[1][0] == '\0') {
+			method = e_met_delete;
+		} else {
+			method = e_met_patch;
+		}
+	}
+	length = strlen(argv) + strlen("/user_data/");
+	if ((path = malloc(length)) == NULL) {
+		return (M_ERROR(1, "Bad alloc"));
+	}
+	path[0] = '\0';
+	strncat(path, "/user_data/", length);
+	strncat(path, argv, length - strlen("/user_data/"));
+    D_REQUEST(init)(&request);
+    if (D_REQUEST(add_host)(&request, g_host, g_port) == false
+        || D_REQUEST(add_user_agent)(&request, "scw-boot-tools/0.1.0") == false
+        || D_REQUEST(add_connection)(&request, "closed") == false
+        || D_REQUEST(add_content_type)(&request, "text/plain") == false
+        || D_REQUEST(add_accept)(&request, "*/*") == false) {
+		sf_free((void **)&path);
+        D_REQUEST(destroy)(&request);
+        return (1);
+    }
+    if (D_CLIENT(init)(&client, g_host, g_port) == false) {
+		sf_free((void **)&path);
+        D_REQUEST(destroy)(&request);
+        return (1);
+    }
+	if (method == e_met_patch) {
+		if (D_REQUEST(add_body)(&request, tab_str[1]) == false) {
+			sf_free((void **)&path);
+			D_REQUEST(destroy)(&request);
+			return (1);
+		}
+	}
+    ret = (D_REQUEST(add_method)(&request, method, path) == false
+        || D_CLIENT(add_data_from_request)(&client, &request) == false
+        || D_CLIENT(send)(&client) == false);
+	if (ret == false) {
+		status_code = uf_get_statuscode(client.v_recv.v_data, client.v_recv.v_length);
+	}
+	if (ret == false && status_code == 200) {
+		dprintf(1, "%s\n", uf_get_body(client.v_recv.v_data, client.v_recv.v_length));
+	}
+	sf_free((void **)&path);
+    D_CLIENT(destroy)(&client);
+    D_REQUEST(destroy)(&request);
+	if (status_code != 200 && status_code != 204) {
+		dprintf(2, "REQUEST ERROR [%d]\n", status_code);
+	}
+    return (!(status_code == 200 || status_code == 204));
+}
+
+int
+main(int ac, const char **av) {
+    int                     ch;
+    static struct option    long_options[] = {
+                            {"state", required_argument, NULL, 's'},
+                            {"userdata", required_argument, NULL, 'u'},
+                            {0, 0, 0, 0}
+                        };
+
+    mlockall(MCL_CURRENT | MCL_FUTURE);
+    while ((ch = getopt_long(ac, (char * const *)av, "s:u:", long_options, NULL)) != -1) {
+        switch (ch) {
+            case 's':
+                return (uf_state(optarg));
+            case 'u':
+                return (uf_userdata(optarg));
+        }
+    }
+    dprintf(2, "Usage: %s [ OPTIONS ] params\n   -s STATE\n   -u KEY=VALUE\n", av[0]);
```

Comment:
`dprintf(2, "Usage: %s [ OPTIONS ] params\n   -s STATE\n   -u KEY=VALUE\n", av[0]);`

should be

`dprintf(2, "Usage: %s [ OPTIONS ] params\n   -s STATE\n   -u KEY[=[VALUE]]\n", av[0]);`